wallet-test/src/test/scala/org/bitcoins/wallet/BitcoindBackendTest.scala

```diff
@@ -0,0 +1,81 @@
+package org.bitcoins.wallet
+
+import org.bitcoins.commons.jsonmodels.wallet.SyncHeightDescriptor
+import org.bitcoins.core.currency._
+import org.bitcoins.db.AppConfig
+import org.bitcoins.server.{BitcoinSAppConfig, BitcoindRpcBackendUtil}
+import org.bitcoins.testkit.fixtures.BitcoinSFixture
+import org.bitcoins.testkit.util.BitcoinSAsyncTest
+import org.bitcoins.testkit.wallet.BitcoinSWalletTest
+import org.bitcoins.testkit.{BitcoinSTestAppConfig, EmbeddedPg}
+import org.bitcoins.wallet.config.WalletAppConfig
+
+import scala.concurrent.Await
+import scala.concurrent.duration.DurationInt
+
+class BitcoindBackendTest extends BitcoinSAsyncTest with EmbeddedPg {
+
+  /** Wallet config with data directory set to user temp directory */
+  implicit protected def config: BitcoinSAppConfig =
+    BitcoinSTestAppConfig.getNeutrinoWithEmbeddedDbTestConfig(pgUrl)
+
+  implicit protected def walletAppConfig: WalletAppConfig = {
+    config.walletConf
+  }
+
+  override def beforeAll(): Unit = {
+    AppConfig.throwIfDefaultDatadir(config.walletConf)
+    super[EmbeddedPg].beforeAll()
+  }
+
+  override def afterAll(): Unit = {
+    Await.result(config.chainConf.stop(), 1.minute)
+    Await.result(config.nodeConf.stop(), 1.minute)
+    Await.result(config.walletConf.stop(), 1.minute)
+    super[EmbeddedPg].afterAll()
+  }
+
+  it must "correctly catch up to bitcoind" in {
+    val amountToSend = Bitcoins.one
+
+    for {
+      // Setup bitcoind
+      bitcoind <- BitcoinSFixture.createBitcoindWithFunds()
+      header <- bitcoind.getBestBlockHeader()
+
+      // Setup wallet
+      tmpWallet <-
+        BitcoinSWalletTest.createDefaultWallet(bitcoind, bitcoind, None)
+      wallet =
+        BitcoindRpcBackendUtil.createWalletWithBitcoindCallbacks(bitcoind,
+                                                                 tmpWallet)
+      // Assert wallet is empty
+      isEmpty <- wallet.isEmpty()
+      _ = assert(isEmpty)
+
+      // Send to wallet
+      addr <- wallet.getNewAddress()
+      _ <- bitcoind.sendToAddress(addr, amountToSend)
+      _ <- bitcoind.generateToAddress(6, addr)
+
+      // assert wallet hasn't seen it yet
+      firstBalance <- wallet.getBalance()
+      _ = assert(firstBalance == Satoshis.zero)
+
+      // Set sync height
+      _ <-
+        wallet.stateDescriptorDAO.updateSyncHeight(header.hashBE, header.height)
```

Comment:
I'm confused, why are we _externally_ calling `updateSyncHeight()` here? Shouldn't this be set internally when we call `catchupWalletToBitcoind()`. Does this have to be done because of the comment I left in the case of syncing the wallet that only has unconfirmed utxos when the bitcoin-s wallet is shutdown?

Reply:
If this wasn't set it would be considered a new wallet and not try to catch up. This is just so we are able to test the catch up sequence.

Reply:
If this wasn't set it would be considered a new wallet and not try to catch up. This is just so we are able to test the catch up sequence.